examples/benchmark/models.py

```diff
@@ -0,0 +1,184 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+import abc
+import os
+import random
+import time
+from functools import cached_property
+from io import IOBase
+from typing import Optional, Any, Tuple, Union
+
+import torch
+import torch.nn as nn
+from PIL import Image
+from omegaconf import DictConfig
+from torch.utils.data.dataloader import DataLoader
+from torchvision.transforms import v2
+from transformers import ViTForImageClassification
+
+from benchmark_utils import ExperimentResult
+from s3torchconnector import S3Reader, S3Checkpoint
+
+
+class ModelInterface(metaclass=abc.ABCMeta):
+    def __init__(self, name: str):
+        self.name = name
+
+    def load_sample(self, sample: Union[S3Reader, Tuple[str, IOBase]]):
+        """Transform given sample (a file-like Object) to a model's input"""
+        if isinstance(sample, S3Reader):
+            key, data = sample.key, sample
+        else:
+            key, data = sample
+
+        return key, data
+
+    @abc.abstractmethod
+    def train_batch(self, batch_idx: int, data, target) -> Optional[Any]:
+        raise NotImplementedError
+
+    def train(self, dataloader: DataLoader, epochs: int) -> ExperimentResult:
+        """Train the model using given dataloader for number of epochs"""
+        num_samples = 0
+        start_time = time.perf_counter()
+        checkpoint_results = []
+        for epoch in range(0, epochs):
+            for batch_idx, (data, target) in enumerate(dataloader):
+                result = self.train_batch(batch_idx, data, target)
+                num_samples += len(data)
+                if result:
+                    checkpoint_results.append(result)
+        end_time = time.perf_counter()
+        training_time = end_time - start_time
+
+        return ExperimentResult(
+            training_time=training_time,
+            num_samples=num_samples,
+            checkpoint_results=checkpoint_results,
+        )
+
+    @abc.abstractmethod
+    def save(self, **kwargs):
+        """Save checkpoint"""
+        raise NotImplementedError
+
+
+class Entitlement(ModelInterface):
+    """
+    This is not really a training model as it does not train anything. Instead, this model simply reads the binary
+    object data from S3, so that we may identify the max achievable throughput for a given dataset.
+    """
+
+    def __init__(self, num_labels: int = None):
+        super(Entitlement, self).__init__("Entitlement")
```

Comment:
The name attribute here is always the classes name. Why not use `type(self).__name__` in the parent/usages?